doc/dox/dev/expressions/questions.md

```diff
@@ -0,0 +1,15 @@
+# Expression Layer Questions
+
+This is just a collection of random musings/questions about the expression
+layer.
+
+Why do the engines not obey RAII? *i.e.* why all the init functions?
+
+Should `init_vars`, `init_struct`, and `init_distribution` really shadow their
+base class versions?
+
+Why have the override pointer propagate the entire expression stack vs. having
+the override occur at the top?
+
+Why is permutation not part of the expression layer? Would decouple much of the
```

Comment:
you would still have to fuse nodes of the tree so the rewriting logic would still need to deal with the permutation fusion (the engine hierarchy is effectively rewritten (at runtime) AST)

Reply:
I actually didn't mean to check this file in (It's not even referenced from any other file). I can remove it (unless you want it for some reason).